pkg/operator/controller/ingress/deployment.go

```diff
@@ -941,6 +951,9 @@ func hashableDeployment(deployment *appsv1.Deployment, onlyTemplate bool) *appsv
 	// Copy metadata and spec fields to which any changes should trigger an
 	// update of the deployment but should not trigger a rolling update.
 	hashableDeployment.Labels = deployment.Labels
+	if deployment.Spec.MinReadySeconds != 30 {
+		hashableDeployment.Spec.MinReadySeconds = deployment.Spec.MinReadySeconds
```

Comment:
What is the purpose of the condition?  This will cause a deployment with `spec.minReadySeconds: 0` and a deployment with `spec.minReadySeconds: 30` to hash to the same value (if all other fields are the same), which would prevent the operator from updating the deployment if it had `spec.minReadySeconds: 0`.